### PR TITLE
Support completing attribute shortcut labels via Context Portal

### PR DESCRIPTION
**Pull request by mgramigna**
_Pull requests into Flux require the following. Submitter should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable (**N/A**) and :white_check_mark:. Reviewers should complete every item on the bulleted list when reviewing._

Can now complete attribute labels (e.g. #status). Currently, the functionality works similar to something like @condition. You can type #status in the editor and then the picker portal will appear to select a value. I did this by adding a picker option to the shortcuts in `Shortcuts.json` as well as update the triggers we manually set in `ShortcutManager` to use said picker.

Things not covered that will be covered by other tasks:

* Completion of attributes that support multiple options (e.g. #reasons)
* Ability to close the context portal and re-open for selection when your cursor is next to an incomplete attribute.

Test case:

1) type #disease status (should show up as incomplete)
2) type #status (#disease status should now be complete, #status should be incomplete and context portal should open)
3) Select a value, should replace #status with the selected value.

## Submitter:

**Running the Application**

- [x] Manually tested in Chrome (primary browser for testing)
- [ ] Manually tested in IE/Safari (verify app loads and basic feature testing)
- [ ] Manually tested in Chrome in iPad mode (at minimum, viewing patient record)

**Task**

- [x] Branch implements task as expected
- [ ] Key application features are updated as needed for task and tested (see: [Key Application Features](https://github.com/FluxNotes/flux/wiki/Key-Application-Features))
- [x] Task specific definition of done (as described in the Jira task) is met, if applicable.

**Documentation**

- [x] This pull request describes why these changes were made
- [x] Recognizes any potential shortcomings/bugs in the description 
- [ ] Pre release script is updated on the [Pre Release Testing Script wiki page](https://github.com/FluxNotes/flux/wiki/Pre-Release-Testing-Script) (add/remove single features to test, do not run through full script)
- [ ] Update the list of key features with new features on the [Key Application Features wiki page](https://github.com/FluxNotes/flux/wiki/Key-Application-Features)
- [ ] Document any new APIs/extension points
- [ ] Additional technical components and libraries are documented and added to [wiki](https://github.com/FluxNotes/flux/wiki/About-Flux-Notes) as needed

**Code Quality**

- [x] Code style guide is followed (see: [Code Style Guide](https://github.com/FluxNotes/flux/wiki/JavaScript-and-HTML-Code-Style-Guide))
- [x] Code is commented

**Tests**

- [x] Existing tests passed
- [ ] Added backend tests for new functionality added

## Reviewers:

- Check that code is maintainable and reusable. Code reuses existing code and infrastructure where appropriate.
- Check the PR and code accomplishes the task's purpose.
- Check that new tests appropriately test the new code, including edge cases. Other existing tests pass.
- Verify Pull Request checklist is correctly completed by submitter.
- Try to break the code

### CODE REVIEW

**Comment by jafeltra** [comment]
This looks cool! One thing I noticed:

[JA1]: When I type #disease status #reasons, before picking a choice, reasons doesn't show up as incomplete anymore. It does on the base branch.

**Comment by sgober** [comment]
[SO1] When selecting a reason, I am seeing a rogue comma show up in the TDP before the reason I selected.

**Comment by mgramigna** [comment]
Resolved JA1 and SO1. They were the same root cause, and I was able to just delete the calls to `setText(<initiating trigger>)` since we don't need to do that anymore after @karenzheung 's refactor. This was causing the attribute value for #reasons to be set prematurely.

**Comment by mgramigna** [comment]
Re: JA2

This happens because of the call to `shortcut.onBeforeDeleted()` on line 402 of `FluxNotesEditor`, which has been in there for a while. This gets triggered when you hit Esc to close the context portal when it is open. This only used to happen for Inserters, but now that we have a picker for CreatorChild shortcuts, we go into this logic too.

Part of this logic removes the shortcut that you are selecting a value for from its parent context's array of children. This reason it removes the shortcut from the children when you hit Esc is because it is added to that array of children every time `newCurrentShortcut` is called in `FullApp` which is called when selecting from the suggestion portal before the context portal even opens.

Now that we actually has code that does something with `this.children` for something like #disease status, we encounter a case where we don't want `onBeforeDeleted` to be called for something like #status because we still want it to be in #disease status' array of children even if it doesn't have a value, as that is what determines the incompleteness of #disease status. We can't just remove it though, because we can then encounter a case where if act crazy enough you could add infinitely many shortcuts to `this.children` if you repeatedly choose from the suggestion portal and then immediately close the Context Portal.

I think this might be more of an issue with the `shortcut-entry-overhaul` branch as a whole rather than this PR, but I am open to more discussion about this tomorrow.

**TL;DR: Existing behavior is causing this that we never caught before we had a label approach, might want to think about resolving it outside of this PR instead of a JA**

**Comment by Dtphelan1** [comment]
I think the suggestion to make this a separate JIRA makes sense to me. The root problem exists on the overhaul branch separately from this, will impact other features beyond just this, and falls outside the immediate scope of this functionality. That's my two sense.

**Comment by jafeltra** [comment]
I agree with both comments, and I think it makes sense to add a Jira that will be part of the Shortcut Guidance epic since it's not directly related to this PR/task.
